src/main/java/net/pterodactylus/fcp/highlevel/FcpClient.java

```diff
@@ -322,6 +286,206 @@ public GetResult getURI(final String uri) throws IOException, FcpException {
 		return getURI(uri, true);
 	}
 
+	/**
+	 * TODO: Turn this into a DDAHandshake function where read and write can be specified through arguments. Handle errors better.
+	 * If we haven't established DDA for a specific directory, do that.
+	 * @param directory - The directory you want to establish Direct-Disk Access for.
+	 *
+	 * @return
+	 * @throws IOException
+	 * @throws FcpException
+	 */
+	public void DDAReadHandshake(final String directory) throws IOException, FcpException{
+
+		new PhageFCPClient.ExtendedAdapter(){
```

Comment:
It’s been quite a while since I last looked at this project but I am almost 100% sure that I don’t know who or what `PhageFCPClient` is… :)

Reply:
I Will clean this up a bit more. Sorry for the sloppy PR